Assets/Tests/Runtime/ConnectionManagementTests.cs

```diff
@@ -0,0 +1,593 @@
+
+            yield return ConnectClients();
+            for (var i = 0; i < NumberOfClients; i++)
+            {
+                Assert.IsTrue(m_ClientNetworkManagers[i].IsConnectedClient);
+            }
+
+            // wait for client synchronization to be over server-side before shutting down client
+            yield return null;
+            yield return null;
+
+            // Disconnecting the client at the transport level
+            m_ClientNetworkManagers[0].NetworkConfig.NetworkTransport.DisconnectLocalClient();
+
+            // Waiting until shutdown is complete
+            yield return new WaitWhile(() => m_ClientNetworkManagers[0].ShutdownInProgress);
+            Assert.IsFalse(m_ClientNetworkManagers[0].IsConnectedClient);
+
+            // Waiting for client to automatically reconnect
+            yield return WaitForClientsConnectedOrTimeOut();
+            Assert.IsTrue(m_ClientNetworkManagers[0].IsConnectedClient);
+
+            var expectedServerConnectionStateSequence = new List<ConnectionState>();
+            expectedServerConnectionStateSequence.Add(m_ServerConnectionManager.m_StartingHost);
+            expectedServerConnectionStateSequence.Add(m_ServerConnectionManager.m_Hosting);
+            Assert.AreEqual(expectedServerConnectionStateSequence, m_ServerConnectionStateSequence);
+
+            for (var i = 0; i < NumberOfClients; i++)
+            {
+                var expectedClientConnectionStateSequence = new List<ConnectionState>();
+                expectedClientConnectionStateSequence.Add(m_ClientConnectionManagers[i].m_ClientConnecting);
+                expectedClientConnectionStateSequence.Add(m_ClientConnectionManagers[i].m_ClientConnected);
+                if (i == 0)
+                {
+                    expectedClientConnectionStateSequence.Add(m_ClientConnectionManagers[i].m_ClientReconnecting);
+                    expectedClientConnectionStateSequence.Add(m_ClientConnectionManagers[i].m_ClientConnected);
+                }
+                Assert.AreEqual(expectedClientConnectionStateSequence, m_ClientConnectionStateSequences[i]);
+            }
+
+        }
+
+        [UnityTest]
+        public IEnumerator UnexpectedServerShutdown_ClientsFailToReconnect()
+        {
+            StartHost();
+            Assert.IsTrue(m_ServerNetworkManager.IsHost);
+
+            SetUniqueProfilesForEachClient();
+
+            yield return ConnectClients();
+            for (var i = 0; i < NumberOfClients; i++)
+            {
+                Assert.IsTrue(m_ClientNetworkManagers[i].IsConnectedClient);
+            }
+
+            // Shutting down the server
+            m_ServerNetworkManager.Shutdown();
+            // ConnectionManager should also transition to Offline state normally, but OnClientDisconnectCallback is not
+            // invoked on the host when shutting down, so transitioning here manually
+            m_ServerConnectionManager.ChangeState(m_ServerConnectionManager.m_Offline);
```

Comment:
If this doesn't happen boss room side, why are we doing it here?
This seems like cheating. It looks like it's related to the issue where a host doesn't get any callbacks when relay disappears, is that it? I wouldn't workaround this here. I'd add some comment on this and create a test that fails and add a [Ignore] attribute on it to keep track of it.

Reply:
Yes that is it. Adding the [Ignore] attribute instead of this workaround.